app/forms/otp_delivery_selection_form.rb

```diff
@@ -15,7 +15,7 @@ def initialize(user, phone_to_deliver_to, context)
 
   def submit(params)
     self.otp_delivery_preference = params[:otp_delivery_preference]
-    self.resend = params[:resend]
+    self.resend = params[:resend] == 'true'
```

Comment:
my fave trick here is:
```suggestion
    self.resend = params[:resend].to_s == 'true'
```
because sometimes when we controller test, or use JSON serialization, the params get passed as literal booleans, so this is a little more forgiving since it works on `"true"` and also `true`